id: dsq-747515442
date: 2007-04-24T07:28:11.0000000-07:00
name: Al
avatar: https://disqus.com/api/users/avatars/Al.jpg
message: "<p>Hi Phil, thanks for the post. I also started using RM for mock tests. I hit a snag though. Running the test throws this \"Error1TestCase 'AS.Com.Business.ProjectSpace.Tests.ProjectAddViewPresenterTests.AddNewProject'<br>failed: System.Reflection.TargetInvocationException : Exception has been thrown by the target of an invocation.<br>  ----&gt; System.NullReferenceException : Object reference not set to an instance of an object.<br>This is my view:<br>---------------------------<br>public interface IProjectAddView<br>{<br>Project Project { get; }<br>event EventHandler ProjectAdded;<br>}<br>This is my presenter:<br>-----------------------------<br>public class ProjectAddViewPresenter<br>{<br>private IProjectAddView _view;<br>public ProjectAddViewPresenter(IProjectAddView view)<br>        {<br>            if (null == view)<br>            {<br>                throw new PresenterException(\"IProjectView can not be null.\");<br>            }<br>            _view = view;<br>            SubscribeToViewEvents();<br>        }<br><br>        public void SubscribeToViewEvents()<br>        {<br>            _view.ProjectAdded += new EventHandler(OnProjectCreated);<br>        }<br>        void OnProjectCreated(object sender, EventArgs args)<br>        {<br>            ProjectManager.CreateProject(_view.Project);<br>        }<br>    }<br><br>This is my test:<br>-----------------------------<br>[Test]<br>        public void AddNewProject()<br>        {<br>            MockRepository mocks = new MockRepository();<br>            IProjectAddView viewMock = (IProjectAddView)mocks.CreateMock(typeof(IProjectAddView));<br>            viewMock.ProjectAdded += null;<br>            IEventRaiser raiser = LastCall.IgnoreArguments().GetEventRaiser();<br>            mocks.ReplayAll();<br>            ProjectAddViewPresenter presenter = new ProjectAddViewPresenter(viewMock);<br>            raiser.Raise(viewMock, EventArgs.Empty);<br>            mocks.VerifyAll();<br>        }<br>Do you have any idea why is the exception? I am sorry for such long post but I can not resolve this. Thanks a bunch in advance.</p>"
